auth/tasks.py

```diff
@@ -50,51 +64,77 @@ def update_group_and_perms(department, created):
         if created:
             assign_model_perms_for_department(department)
 
+    def remove_related_department_group(super_department, childs):
+        # 将当前department的所有叶子结点返回
+        child_departments = Department.child_departments.all()
+        if child_departments:
+            for child_department in child_departments:
+                childs.extend(
```

Comment:
Every time you `.extend()` the result of `remove_related_department_group()`, you are duplicating its content, please double-check the logic matches your intention.